packages/browser/test/unit/transports/fetch.test.ts

```diff
@@ -104,6 +107,32 @@ describe('FetchTransport', () => {
       }
     });
 
+    it('should record dropped event when fetch fails', async () => {
+      const response = { status: 403, headers: new Headers() };
+
+      window.fetch.mockImplementation(() => Promise.reject(response));
+
+      const spy = jest.spyOn(transport, 'recordLostEvent');
+
+      try {
+        await transport.sendEvent(eventPayload);
+      } catch (_) {
+        expect(spy).toHaveBeenCalledWith(Outcome.NetworkError, 'event');
+      }
```

Comment:
Should we also expect the reason the promise got rejected?

Reply:
We do that in one test earlier